src/composables/use-scroll.js

```diff
@@ -0,0 +1,43 @@
+// code taken from Vueuse
+import throttle from 'lodash.throttle'
+import { ref } from '@nuxtjs/composition-api'
+import { useEventListener } from '~/composables/use-event-listener'
+
+/**
+ *
+ * @param { import('./types').MaybeRef<HTMLElement> } element
+ * @param {object} options
+ * @param {number} [options.throttleMs] - time to throttle the scroll handler.
+ * Set to 0 to remove throttling
+ */
+export function useScroll(element, { throttleMs = 200 } = {}) {
+  /** @type { import('@nuxtjs/composition-api').Ref<number> } */
+  const y = ref(0)
+  /**
+   * Whether the element has been scrolled down at all
+   *
+   * @type { import('@nuxtjs/composition-api').Ref<boolean> }
+   */
+  const isScrolled = ref(false)
+
+  if (element) {
+    /** @param { Event } e */
+    const scrollHandler = (e) => {
+      const eventTarget =
+        e.target === document ? e.target.documentElement : e.target
+      y.value = eventTarget.scrollTop
+      isScrolled.value = y.value > 0
+    }
+
+    const handler = throttleMs
+      ? throttle(scrollHandler, throttleMs)
+      : scrollHandler
+
+    useEventListener(element, 'scroll', handler, {
+      capture: false,
+      passive: true,
+    })
+  }
+
+  return { y, isScrolled }
```

Comment:
Their purposes are similar. At first, we were scrolling the whole window, but then we updated the layout so that it consists of a grid, in which the main content and the sidebar can be scrolled separately. This means that we can no longer detect if the header has scrolled or not by using `window`'s scroll offset, and need to get the main content `scrollTop` instead. I did not remove the `use-window-scroll.js` just yet. But if it turns out after all the redesign work that we don't, we should delete `use-window-scroll.js`.